mipengine/controller/api/validator.py

```diff
@@ -0,0 +1,312 @@
+    try:
+        algorithm_request = AlgorithmRequestDTO.from_json(request_body)
+    except Exception:
+        logging.error(
+            f"Could not parse the algorithm request body. "
+            f"Exception: \n {traceback.format_exc()}"
+        )
+        raise BadRequest(f"The algorithm request body does not have the proper format.")
+
+    algorithm_specs = _get_algorithm_specs(algorithm_name)
+    _validate_algorithm_parameters(algorithm_specs, algorithm_request)
+
+
+def _get_algorithm_specs(algorithm_name):
+    if algorithm_name not in algorithms_specifications.enabled_algorithms.keys():
+        raise BadRequest(f"Algorithm '{algorithm_name}' does not exist.")
+    return algorithms_specifications.enabled_algorithms[algorithm_name]
+
+
+def _validate_algorithm_parameters(
+    algorithm_specs: AlgorithmSpecifications, algorithm_request: AlgorithmRequestDTO
+):
+    _validate_inputdata(algorithm_specs.inputdata, algorithm_request.inputdata)
+
+    _validate_generic_parameters(
+        algorithm_specs.parameters, algorithm_request.parameters
+    )
+
+
+def _validate_inputdata(
+    inputdata_specs: InputDataSpecifications, input_data: AlgorithmInputDataDTO
+):
+    _validate_pathology_and_dataset_values(input_data.pathology, input_data.datasets)
+
+    _validate_inputdata_filter(input_data.filters)
+
+    _validate_inputdata_cdes(inputdata_specs, input_data)
+
+
+def _validate_pathology_and_dataset_values(pathology: str, datasets: List[str]):
+    """
+    Validates that the pathology, dataset values exist and
+    that the datasets belong in the pathology.
+    """
+
+    if not node_catalog.pathology_exists(pathology):
+        raise BadUserInput(f"Pathology '{pathology}' does not exist.")
+
+    # TODO Remove with pydantic
+    if not isinstance(datasets, list):
+        raise BadRequest(f"Datasets parameter should be a list.")
+
+    if not all(node_catalog.dataset_exists(pathology, dataset) for dataset in datasets):
+        raise BadUserInput(
+            f"Datasets '{datasets}' do not belong in pathology '{pathology}'."
+        )
+
+
+def _validate_inputdata_filter(filter):
+    """
+    Validates that the filter provided have the correct format
+    following: https://querybuilder.js.org/
+    """
+    # TODO Add filter
+    pass
+
+
+def _validate_inputdata_cdes(
+    input_data_specs: InputDataSpecifications, input_data: AlgorithmInputDataDTO
+):
+    _validate_inputdata_cde(input_data_specs.x, input_data.x, input_data.pathology)
+    _validate_inputdata_cde(input_data_specs.y, input_data.y, input_data.pathology)
+
+
+def _validate_inputdata_cde(
+    cde_parameter_specs: InputDataSpecification,
+    cde_parameter_value: Optional[List[str]],
+    pathology: str,
+):
+    if cde_parameter_specs.notblank and not cde_parameter_value:
+        raise BadUserInput(
+            f"Inputdata '{cde_parameter_specs.label}' should be provided."
+        )
+
+    if not cde_parameter_value:
+        return
```

Comment:
Refactored the
```
if not cde_parameter_value:
    return
```
to make it more clear.

About the "common data elements" I understand what you mean. I suppose this will be dealt with, when the engine/application separation occurs.